### DISCUSSION TITLE
How to add another login plugin?

### DISCUSSION THREAD

**Discussion by wtlgo** [games647/FastLogin#789] [Q&A] [answered]
Hello! I am looking through the source code and can't figure out how to add a hook for my own (currently private) auth plugin without forking and modifying the FastLogin project. How do I do this properly?

**Comment by TuxCoding** [comment, marked as answer]
This functionality is shared between all three platforms. Therefore it's inside the core component:

https://github.com/games647/FastLogin/blob/main/core/src/main/java/com/github/games647/fastlogin/core/shared/FastLoginCore.java#L288

**Comment by wtlgo** [comment]
So what do I do? Should I call this method with my own hook object inherited from `AuthPlugin<P>` when my plugin inits, is that right?

**Reply by TuxCoding** [reply]
Yes, your plugin has to add a softdepend on FastLogin and then register your own hook using the given method.

**Reply by wtlgo** [reply]
@games647 oh cool, thank you!
